index.rst

```diff
@@ -2,24 +2,180 @@
+
+The most obvious advantage of using regular operations for integration testing is that we are developing and using SAL Script that will be regularly used.
+Most of these operations exercise the vast majority of the system requirements, which makes them a perfect tool to verify them.
+Since the majority of these uses TSSW high-level libraries (e.g. ts_observatory_control), they are extremely maintainable, requiring minimal to no updates when there are underlying changes to the CSCs behavior.
+
+Jupyter Notebooks
+-----------------
+
+Jupyter notebooks provides a complimentary interactive interface into the Rubin Observatory Control system.
+They are used in parallel with SAL Scripts to execute tasks that require some user interaction or a finer degree of control over the operations.
+
+Although they can be very useful in the early development stages of the process, we do not recommend storing any long-term use software in notebooks.
+The main reason is that it is currently note possible to unit test Jupyter notebooks, which can leave to bit rot in short timescales, especially with the rapidly evolving software development one would expect in early commissioning stages.
+
+Exceptional Cases
+-----------------
+
+.. Since part of Telescope and Site Software scope is to deliver a CI/CD system for the observatory control software, this system
+
+Although a substantial number of requirements can be verified with the use of our CI system, we acknowledge that some of them will inevitably demand a more individualized approach and, sometimes, even a more involved process demanding additional data analysis and/or a more descriptive verification methodology.
+
+Ideally, these cases would be addressed by either developing a customized SAL Script to drive the system through the required test steps or, in the more extreme conditions that might require interactive steps, a Jupyter notebook.
+
+Furthermore, any detailed data analysis/methodology should be included in a technical note to be incorporated into the requirement verification artifacts.
+
+.. , since we execute (or plan to) these tests continuously, we can verify that the system continues to fullfil requirements at every iteration.
+
+.. Because these functionalities are (mostly) taken from the requirements pool, it means that each new feature is generally accompanied by a requirement verification test.
+.. Nevertheless, tests comes in a variety of different shapes and forms and, most importantly, sometimes it is not possible to fully exercise a given requirement in a testing environment.
+
+.. _verification-artifacts:
+
+Verification artifacts
+======================
+
+In general, successfully executing the testing suite from start to finish gives us confidence that the system is operating as expected.
```

Comment:
happens

Reply:
uhm... can you clarify? 🤔

Reply:
I have absolutely no idea what this is... sorry.